tests/acceptance/infinity-route-with-embedded-route-test.js

```diff
@@ -1,31 +0,0 @@
-import { test } from 'qunit';
-import moduleForAcceptance from '../../tests/helpers/module-for-acceptance';
-import buildServer from '../helpers/fake-album-server';
-
-let server;
-
-moduleForAcceptance('Acceptance: Infinity Route', {
-  beforeEach() {
-    server = buildServer();
-  },
-  afterEach() {
-    server.shutdown();
-  }
-});
-
-test('it works when embedded route is refreshed', function(assert) {
-  visit('/posts/1');
-  click('button.refreshRoute');
-
-  andThen(() => {
-    assert.equal(find('ul').find('li').length, 2);
-
-    var testList = find('ul');
-    testList.scrollTop(2000);
-    triggerEvent('ul', 'scroll');
-
-    andThen(() => {
-      assert.equal(find('ul').find('li').length > 2, true);
-    });
-  });
-});
```

Comment:
Why did we remove this test?

Reply:
To me it seemed to be asserting the same behavior as the other acceptance tests.  I.e. even though clicking the button, we are still scrolling down the page to trigger loading more items.

Reply:
Oh right.  Historical context:  Given the way Ember Infinity used to fire actions, there was an issue when you had a nested route  (ie: posts -> posts.index) with an infinity collection on both routes:

- POSTS
    -- categories (infinity)
    -- POSTS SHOW
      -- posts (infinity)

The issue was that if you fired an action on `posts (infinity)` it would bubble up and load more categories.

This test is just asserting that the fix we put in place back then is working: the right collections are loaded on the right routes when the action is triggered.

Reply:
@hhff So thinking this test is still needed?

Reply:
Yes, please!

Reply:
Ok, I think I'll have to make a PR to ember-in-viewport.  Right now the scrollable area is the window.  In this test, it looks like the scrollable area is contained.

Reply:
Yeah - that's correct.  In order for two routes to be independently infinity-scrollable, at least one of them needs to be an `overflow:hidden` container.

The changes made here should work in `overflow:hidden` containers though right?

Reply:
Well I think it is making the fetch too early.  So the test runner window is the browser window.  The viewport for the list of posts is only a fraction of the window.  So the infinite loader, once it comes into view of the window (not the scrollable area), it is making a request.

So we want the component only to make a request if it is in view of the scrollable area, not the window.  Lmk if that makes sense...

Reply:
@hhff Added it back.

Reply:
NICE

Reply:
looks like that fixed the other test (checking for 2 records rather than 3) also?

Reply:
Yeah I was wrong about the CSS thing.  It was literally the lack of a scrolling container. :+1:

Reply:
beautiful - glad we caught that!